File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.js
/**
 * @param {string} s
 * @param {number} k
 * @return {boolean}
 */
var canConstruct = function (s, k) {
    const n = s.length;
    
    if (n < k) {
        return false;
    }

    let odd = 0;
    let mp = new Map();

    for (let ch of s) {
        mp.set(ch, (mp.get(ch) || 0) + 1);
    }

    for (let [key, val] of mp) {
        if (val % 2 != 0) {
            odd++;
        }
    }

    if (n == k || odd <= k) {
        return true;
    }
    return false;
};

File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) == k:
            return True
        if len(s) < k:
            return False
        counter = Counter(s)
        odd = 0

        for key, val in counter.items():
            if val % 2 != 0:
                odd += 1

        return odd <= k